tests/baselines/reference/derivedClassSuperProperties.js

```diff
@@ -0,0 +1,844 @@
+    __extends(DerivedWithDecoratorOnClassMethod, _super);
+    function DerivedWithDecoratorOnClassMethod() {
+        var _this = this;
+        _this.prop = true;
+        var InnerClass = /** @class */ (function () {
+            function InnerClass() {
+            }
+            InnerClass.prototype.innerMethod = function () { };
+            __decorate([
+                decorate(this)
+            ], InnerClass.prototype, "innerMethod", null);
+            return InnerClass;
+        }());
+        _this = _super.call(this) || this;
+        return _this;
+    }
+    return DerivedWithDecoratorOnClassMethod;
+}(Base));
+var DerivedWithDecoratorOnClassProperty = /** @class */ (function (_super) {
+    __extends(DerivedWithDecoratorOnClassProperty, _super);
+    function DerivedWithDecoratorOnClassProperty() {
+        var _this = this;
+        _this.prop = true;
+        var InnerClass = /** @class */ (function () {
+            function InnerClass() {
+                this.innerProp = true;
+            }
+            __decorate([
+                decorate(this)
+            ], InnerClass.prototype, "innerProp", void 0);
+            return InnerClass;
+        }());
+        _this = _super.call(this) || this;
+        return _this;
+    }
+    return DerivedWithDecoratorOnClassProperty;
+}(Base));
+var DerivedWithFunctionDeclaration = /** @class */ (function (_super) {
+    __extends(DerivedWithFunctionDeclaration, _super);
+    function DerivedWithFunctionDeclaration() {
+        var _this = this;
+        _this.prop = true;
+        function declaration() {
+            return this;
+        }
+        _this = _super.call(this) || this;
+        return _this;
+    }
+    return DerivedWithFunctionDeclaration;
+}(Base));
+var DerivedWithFunctionDeclarationAndThisParam = /** @class */ (function (_super) {
+    __extends(DerivedWithFunctionDeclarationAndThisParam, _super);
+    function DerivedWithFunctionDeclarationAndThisParam() {
+        var _this = this;
+        _this.prop = true;
+        function declaration(param) {
+            if (param === void 0) { param = this; }
+            return param;
+        }
+        _this = _super.call(this) || this;
+        return _this;
+    }
+    return DerivedWithFunctionDeclarationAndThisParam;
+}(Base));
+var DerivedWithFunctionExpression = /** @class */ (function (_super) {
+    __extends(DerivedWithFunctionExpression, _super);
+    function DerivedWithFunctionExpression() {
+        var _this = this;
+        _this.prop = true;
+        (function () {
+            return this;
+        })();
+        _this = _super.call(this) || this;
+        return _this;
+    }
+    return DerivedWithFunctionExpression;
+}(Base));
+var DerivedWithParenthesis = /** @class */ (function (_super) {
+    __extends(DerivedWithParenthesis, _super);
+    function DerivedWithParenthesis() {
+        var _this = this;
+        _this.prop = true;

```

Comment:
The emit here has the wrong runtime semantics.

Reply:
The emit here is still incorrect. The `_this.prop = true;` statement needs to come after `(_this = _super.call(this) || this);`.